codec-http/src/main/java/io/netty5/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -609,25 +611,26 @@ private State readHeaders(Buffer buffer) {
         HttpMessageDecoderResult decoderResult = new HttpMessageDecoderResult(lineParser.size, headerParser.size);
         message.setDecoderResult(decoderResult);
 
-        List<String> contentLengthFields = headers.getAll(HttpHeaderNames.CONTENT_LENGTH);
-        if (!contentLengthFields.isEmpty()) {
+        Iterator<CharSequence> contentLengthFields = headers.valuesIterator(HttpHeaderNames.CONTENT_LENGTH);
+        boolean hasContentLength = contentLengthFields.hasNext();
+        if (hasContentLength) {
             HttpVersion version = message.protocolVersion();
-            boolean isHttp10OrEarlier = version.majorVersion() < 1 || (version.majorVersion() == 1
-                    && version.minorVersion() == 0);
+            boolean isHttp10OrEarlier = version.majorVersion() < 1 || version.majorVersion() == 1
+                                                                      && version.minorVersion() == 0;
```

Comment:
I know the extra `(...)` is not needed but I still think it makes it easier to read..